implementations/rust/ockam/ockam_identity/src/identity.rs

```diff
@@ -199,9 +207,9 @@ impl<V: IdentityVault, S: AuthenticatedStorage> Identity<V, S> {
     }
 }
 
-impl<V: IdentityVault> Identity<V, InMemoryStorage> {
+impl Identity {
     /// Create an `Identity` with a new secret key and `InMemoryStorage`
-    pub async fn create(ctx: &Context, vault: &V) -> Result<Self> {
+    pub async fn create_arc(ctx: &Context, vault: Arc<dyn IdentityVault>) -> Result<Self> {
```

Comment:
Maybe if `Vault::create()` returns `Arc<Vault>` we could avoid having functions with `_arc` suffix? What do you think?

Reply:
That's a fantastic idea, I was able to remove the 3 `_arc` functions in my latest commit!